File: moi-dotnet/MoiBackend.Core/Entities/GivenMoiEntry.cs
namespace MoiBackend.Core.Entities;

public class GivenMoiEntry
{
    public long Id { get; set; }
    public string RecipientName { get; set; } = null!;
    public string? Village { get; set; }
    public decimal Amount { get; set; }
    public string? GiftType { get; set; } = "Cash"; // "Cash" | "Gold"
    public string? GoldDetails { get; set; }
    public string? Occasion { get; set; } // Function name e.g. Marriage, Ear Piercing
    public string? GiftTerm { get; set; } // 1st Time, 2nd Time, 3rd Time, etc.
    public DateTime GivenDate { get; set; } = DateTime.UtcNow;
    public string? Notes { get; set; }

    public long? EventId { get; set; }
    public Event? Event { get; set; }
}

File: moi_flutter_app/lib/models/given_moi_entry.dart
class GivenMoiEntryModel {
  final int id;
  final String recipientName;
  final String village;
  final String occasion;
  final double amount;
  final DateTime givenDate;
  final String? notes;

  GivenMoiEntryModel({
    required this.id,
    required this.recipientName,
    required this.village,
    required this.occasion,
    required this.amount,
    required this.givenDate,
    this.notes,
  });

  factory GivenMoiEntryModel.fromJson(Map<String, dynamic> json) {
    return GivenMoiEntryModel(
      id: json['id'] ?? 0,
      recipientName: json['recipientName'] ?? json['recipient_name'] ?? json['name'] ?? '',
      village: json['village'] ?? '',
      occasion: json['occasion'] ?? '',
      amount: (json['amount'] is num) ? (json['amount'] as num).toDouble() : 0.0,
      givenDate: json['givenDate'] != null
          ? DateTime.tryParse(json['givenDate']) ?? DateTime.now()
          : DateTime.now(),
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'recipientName': recipientName,
      'village': village,
      'occasion': occasion,
      'amount': amount,
      'givenDate': givenDate.toIso8601String(),
      'notes': notes,
    };
  }
}
